flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -774,6 +855,43 @@ def _iter_nodes(
   for key, value in node_dict.items():
     yield from _iter_nodes(value, visited, (*path_parts, key))
 
+@tp.overload
+def compose_mapping(
+  map_ab: tp.Mapping[A, B],
+  map_bc: tp.Mapping[B, C],
+  /,
+  *,
+  invert: tp.Literal[False] = False,
+) -> dict[A, C]:
+  ...
+
+
+@tp.overload
+def compose_mapping(
+  map_ab: tp.Mapping[A, B],
+  map_bc: tp.Mapping[B, C],
+  /,
+  *,
+  invert: tp.Literal[True],
+) -> dict[C, A]:
+  ...
+
+
+def compose_mapping(
```

Comment:
FYI https://docs.python.org/3/library/typing.html#typing.overload suggests that the implementation should be untyped.

Note sure how many type checkers support that, though.